### PR TITLE
Hide toolbar and select cells on printing

### PR DESCRIPTION
**Pull request by andfs**
When printing, toolbar and select cells are visible and it's a not desired behavior on reports.

### CODE REVIEW

File: src/MUIDataTable.js [hunk 1]
```diff
@@ -241,11 +241,12 @@ class MUIDataTable extends React.Component {
     },
     showResponsive: false,
     searchText: null,
+    isPrinting: false,
   };
 
   constructor() {
     super();
-    this.tableRef = false;
+    this.tableRef = React.createRef();
```
Comment (patorjk) [inline src/MUIDataTable.js:249]: Why is this being changed?

File: src/MUIDataTable.js [hunk 2]
```diff
@@ -259,6 +260,8 @@ class MUIDataTable extends React.Component {
   componentDidMount() {
     this.setHeadResizeable(this.headCellRefs, this.tableRef);
 
+    this.setState({ pagination: this.options.pagination });
```
Comment (patorjk) [inline src/MUIDataTable.js:263]: I don't understand this addition with regards to the printing update.

File: src/MUIDataTable.js [hunk 3]
```diff
@@ -282,6 +285,13 @@ class MUIDataTable extends React.Component {
     }
   }
 
+  isGoingToPrint = () =>
+    new Promise((resolve, reject) => {
+      this.setState({ isPrinting: true, pagination: false }, () => setTimeout(() => resolve(), 500));
+    });
+
+  hasPrintted = () => this.setState({ isPrinting: false, pagination: this.options.pagination });
```
Comment (patorjk) [inline src/MUIDataTable.js:293]: typo: hasPrinted

File: src/MUIDataTable.js [hunk 4]
```diff
@@ -425,7 +435,10 @@ class MUIDataTable extends React.Component {
 
   // must be arrow function on local field to refer to the correct instance when passed around
   // assigning it as arrow function in the JSX would cause hard to track re-render errors
-  getTableContentRef = () => this.tableContent.current;
+  getTableContentRef = () => {
+    this.setState({ isPrinting: true });
+    return this.tableRef ? this.tableRef.current : this.tableRef.current;
```
Comment (patorjk) [inline src/MUIDataTable.js:440]: This always resolves to this.tableRef.current

File: src/MUIDataTable.js [hunk 5]
```diff
@@ -1349,7 +1362,7 @@ class MUIDataTable extends React.Component {
     } = this.state;
 
     const rowCount = this.state.count || displayData.length;
-    const rowsPerPage = this.options.pagination ? this.state.rowsPerPage : displayData.length;
+    const rowsPerPage = this.state.pagination ? this.state.rowsPerPage : displayData.length;
```
Comment (patorjk) [inline src/MUIDataTable.js:1365]: Is there a bug with the pagination?

File: src/MUIDataTable.js (hunk 1 above)
Comment (andfs) [reply src/MUIDataTable.js:249]: This because the update of react and material ui versions

File: src/MUIDataTable.js (hunk 2 above)
Comment (andfs) [reply src/MUIDataTable.js:263]: I forgot to mention it int PR's description, but I'm hiding pagination too. For reports is important to print the whole data, not only the current pagination.

File: src/MUIDataTable.js (hunk 4 above)
Comment (andfs) [reply src/MUIDataTable.js:440]: Yeah, sorry :/

File: src/MUIDataTable.js (hunk 5 above)
Comment (andfs) [reply src/MUIDataTable.js:1365]: Not a bug, but I`m hiding the pagination on printing too

**Comment by andfs** [comment]
I`ll fix those issues and submit a new PR soon

File: src/MUIDataTable.js (hunk 1 above)
Comment (patorjk) [reply src/MUIDataTable.js:249]: ah, I see the 16.3 migration docs on this, you are correct. Thank you for spotting this!

**Comment by andfs** [comment]
Sorry for taking so long...
I fixed the errors in my last commit :)

**Comment by patorjk** [comment]
No worries. Updates for printing should be available in v3, which hopefully will be released within the next day or two (mostly now just need to finish up testing).

**Pull request merged**